src/envs/cover.py

```diff
@@ -312,7 +317,8 @@ def _Covers_holds(state: State, objects: Sequence[Object]) -> bool:
         target_pose = state.get(target, "pose")
         target_width = state.get(target, "width")
         return (block_pose-block_width/2 <= target_pose-target_width/2) and \
-               (block_pose+block_width/2 >= target_pose+target_width/2)
+               (block_pose+block_width/2 >= target_pose+target_width/2) and \
```

Comment:
interestingly, this change makes there be a 4th learned operator in cover when using replays:
```
    Parameters: [?x0:block, ?x1:target]
    Preconditions: [Covers(?x0:block, ?x1:target), HandEmpty(), IsBlock(?x0:block), IsTarget(?x1:target)]
    Add Effects: [Holding(?x0:block)]
    Delete Effects: [Covers(?x0:block, ?x1:target), HandEmpty()]
    Side Predicates: []
```
this corresponds to un-covering something that was already covering, which is now possible via a 1-step deviation from a demo (but wasn't possible before, because you needed to pick it up and then place it elsewhere for Covers to get turned off)

Reply:
good thinking to test this. I'm confused though... wouldn't Holding have been an add effect beforehand too?

Reply:
i see all the test changes though. would you mind just pasting all the operators here?

Reply:
before:
```
STRIPS-Op0:
    Parameters: [?x0:block]
    Preconditions: [HandEmpty(), IsBlock(?x0:block)]
    Add Effects: [Holding(?x0:block)]
    Delete Effects: [HandEmpty()]
    Side Predicates: []
STRIPS-Op1:
    Parameters: [?x0:block, ?x1:target]
    Preconditions: [Holding(?x0:block), IsBlock(?x0:block), IsTarget(?x1:target)]
    Add Effects: [Covers(?x0:block, ?x1:target), HandEmpty()]
    Delete Effects: [Holding(?x0:block)]
    Side Predicates: []
STRIPS-Op2:
    Parameters: [?x0:block]
    Preconditions: [Holding(?x0:block), IsBlock(?x0:block)]
    Add Effects: [HandEmpty()]
    Delete Effects: [Holding(?x0:block)]
    Side Predicates: []
```

after:
```
STRIPS-Op0:
    Parameters: [?x0:block]
    Preconditions: [HandEmpty(), IsBlock(?x0:block)]
    Add Effects: [Holding(?x0:block)]
    Delete Effects: [HandEmpty()]
    Side Predicates: []
STRIPS-Op1:
    Parameters: [?x0:block, ?x1:target]
    Preconditions: [Holding(?x0:block), IsBlock(?x0:block), IsTarget(?x1:target)]
    Add Effects: [Covers(?x0:block, ?x1:target), HandEmpty()]
    Delete Effects: [Holding(?x0:block)]
    Side Predicates: []
STRIPS-Op2:
    Parameters: [?x0:block, ?x1:target]
    Preconditions: [Covers(?x0:block, ?x1:target), HandEmpty(), IsBlock(?x0:block), IsTarget(?x1:target)]
    Add Effects: [Holding(?x0:block)]
    Delete Effects: [Covers(?x0:block, ?x1:target), HandEmpty()]
    Side Predicates: []
STRIPS-Op3:
    Parameters: [?x0:block]
    Preconditions: [Holding(?x0:block), IsBlock(?x0:block)]
    Add Effects: [HandEmpty()]
    Delete Effects: [Holding(?x0:block)]
    Side Predicates: []
```

Reply:
the difference is the delete effects (between Op0 and Op2 in "after")

Reply:
ahhhh okay I get it now, thanks